cmd/fyne_settings/settings/appearance.go

```diff
@@ -128,3 +178,99 @@ func (s *Settings) saveToFile(path string) error {
 
 	return ioutil.WriteFile(path, data, 0644)
 }
+
+type colorButton struct {
+	widget.BaseWidget
+	name    string
+	color   color.Color
+	current bool
+
+	s *Settings
+}
+
+func newColorButton(n string, c color.Color, s *Settings) *colorButton {
+	b := &colorButton{name: n, color: c, s: s}
+	b.ExtendBaseWidget(b)
+	return b
+}
+
+func (c *colorButton) CreateRenderer() fyne.WidgetRenderer {
+	r := canvas.NewRectangle(color.Transparent)
+	r.StrokeWidth = 5
+
+	if c.name == c.s.fyneSettings.PrimaryColor {
+		r.StrokeColor = theme.PrimaryColor()
+	}
+
+	return &colorRenderer{c: c, rect: r, objs: []fyne.CanvasObject{r}}
+}
+
+func (c *colorButton) Tapped(_ *fyne.PointEvent) {
+	c.s.fyneSettings.PrimaryColor = c.name
+	for _, child := range c.s.colors {
+		child.Refresh()
+	}
+
+	c.s.preview.Image = c.s.createPreview()
+	canvas.Refresh(c.s.preview)
+}
+
+type colorRenderer struct {
+	c    *colorButton
+	rect *canvas.Rectangle
+	objs []fyne.CanvasObject
+}
+
+func (c *colorRenderer) Layout(s fyne.Size) {
+	c.rect.Resize(s)
+}
+
+func (c *colorRenderer) MinSize() fyne.Size {
+	return fyne.NewSize(20, 20)
+}
+
+func (c *colorRenderer) Refresh() {
+	if c.c.name == c.c.s.fyneSettings.PrimaryColor {
+		c.rect.StrokeColor = theme.PrimaryColor()
+	} else {
+		c.rect.StrokeColor = color.Transparent
+	}
+
+	c.rect.Refresh()
+}
+
+func (c *colorRenderer) BackgroundColor() color.Color {
+	return c.c.color
+}
+
+func (c *colorRenderer) Objects() []fyne.CanvasObject {
+	return c.objs
+}
+
+func (c *colorRenderer) Destroy() {
+}
+
+func makeHomeTab() fyne.CanvasObject {
```

Comment:
Does this really need to be a separate function?

Reply:
No, it does not have to be.
For a while it contained more content - and in the future it might be more interesting than the simple label.
I can take it out if this is a concern

Reply:
Let's remove for now, we can revisit in future